pytest_splunk_addon/standard_lib/addon_parser/__init__.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+"""
+The module provides the Add-on parsing mechanism. It can
+parse the knowledge objects from an Add-on's configuration files
+
+Supports: fields from props & transforms, tags, eventtypes
+
+Dependencies: 
+    splunk_appinspect.App: To parse the configuration files 
+"""
+import os
+import re
+import logging
+from splunk_appinspect import App
+
+from .fields import convert_to_fields, Field
+from .transforms_parser import TransformsParser
+from .props_parser import PropsParser
+from .tags_parser import TagsParser
+from .eventtype_parser import EventTypeParser
+
+LOGGER = logging.getLogger("pytest-splunk-addon")
```

Comment:
I see this in all the files, but why not put this into the class itself in the init? 

```python
# something like this so that in the logs, you'll see which class is logging 
# what message, instead of having everything log with the same name
def __init__(...): 
    self.logger = logging.getLogger(self.__name__)

Reply:
It's because we are creating a log file for "pytest-splunk-addon" logger which can be used for troubleshooting in CICD pipelines.  If we don't create the logs explicitly, pytest just ignores the log and doesn't add in the Junit-xml file.